cesium/internal/unary/iterator.go

```diff
@@ -318,33 +324,42 @@ func (i *Iterator) read(ctx context.Context, offset telem.Offset, size telem.Siz
 	if n < len(series.Data) {
 		series.Data = series.Data[:n]
 	}
-
 	return
 }
 
-func (i *Iterator) sliceDomain(ctx context.Context) (telem.Offset, telem.Size, error) {
-	startApprox, err := i.approximateStart(ctx)
+func (i *Iterator) sliceDomain(ctx context.Context) (
+	telem.Offset,
+	uint32,
+	telem.Size,
+	error,
+) {
+	startApprox, domain, err := i.approximateStart(ctx)
 	if err != nil {
-		return 0, 0, err
+		return 0, 0, 0, err
 	}
 	endApprox, err := i.approximateEnd(ctx)
 	if err != nil {
-		return 0, 0, err
+		return 0, 0, 0, err
 	}
 	startOffset := i.Channel.DataType.Density().Size(startApprox.Upper)
 	size := i.Channel.DataType.Density().Size(endApprox.Upper) - startOffset
-	return startOffset, size, nil
+	return startOffset, domain, size, nil
 }
 
 // approximateStart approximates the number of samples between the start of the current
 // range and the start of the current iterator view. If the start of the current view is
 // before the start of the range, the returned value will be zero.
-func (i *Iterator) approximateStart(ctx context.Context) (startApprox index.DistanceApproximation, err error) {
+func (i *Iterator) approximateStart(ctx context.Context) (
+	startApprox index.DistanceApproximation,
+	domain uint32,
+	err error,
+) {
+	target := i.internal.TimeRange().Start.SpanRange(0)
 	if i.internal.TimeRange().Start.Before(i.view.Start) {
-		target := i.internal.TimeRange().Start.Range(i.view.Start)
-		startApprox, err = i.idx.Distance(ctx, target, true)
+		target.End = i.view.Start
 	}
-	return
+	sa, da, err := i.idx.Distance(ctx, target, true)
```

Comment:
I understand `sa` and `da`, but I think it's better if we call it `startApprox` and `domainApprox` so we don't need ot think about what `sa` and `da` mean

Reply:
Agreed